crates/foundry-ui/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+pub use message::*;
+
+pub mod message;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum OutputFormat {
+    #[default]
+    Human,
+    Json,
+}
+
+impl OutputFormat {
+    #[must_use]
+    pub fn from_flag(json: bool) -> Self {
+        if json {
+            OutputFormat::Json
+        } else {
+            OutputFormat::Human
+        }
+    }
+}
+
+/// An abstraction around console output which stores preferences for output format (human vs JSON),
+/// colour, etc.
+///
+/// All messaging (basically all writes to `stdout`) must go through this object.
+#[derive(Debug)]
+pub struct UI {
+    output_format: OutputFormat,
+    // TODO(3395): Add state here, that can be used for spinner
+}
+
+impl UI {
+    /// Create a new [`UI`] instance configured with the given output format.
+    #[must_use]
```

Comment:
Now that I think of it, we should probably define a trait for UI and implement it here.

So you can type our function as accepting the UI trait, not the specific implementation of it.
The benefit will be you will be able to mock the UI in tests or provide an implementation that allows us to just capture what is printed so we can assert it (e.g. for tests that show warnings about `snforge_std`).

cc @kkawula @ksew1